### PR TITLE
add README and LICENSE

### PR DESCRIPTION
**Pull request by marcoow**
see #20

### CODE REVIEW

**Comment by marcoow** [comment]
@BobrImperator let me know if you have any feedback as well – mainly interested in whether anything substantial is missing that should be in a README. There will be guides going into much more depth anyway.

File: README.md [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+  drop     Drop the database
+  create   Create the database
+  migrate  Migrate the database
+  reset    Reset (drop, create, migrate) the database
+  seed     Seed the database
+  help     Print this message or the help of the given subcommand(s)
+
+Options:
+  -e, --env <ENV>  Choose the environment (development, test, production). [default: development]
+      --no-color   Disable colored output.
```
Comment (BobrImperator) [inline README.md:198]: I think that the fact that migrations don't run for your tests automatically needs to be explained somewhere i.e. what I've mentioned here #20 

e.g.

> Quickstart:
> pace -f my_server
> cd my_server
> docker compose up -d
> cargo db migrate -e test
> cargo test

File: README.md [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+
+Usage: db [OPTIONS] <COMMAND>
+
+Commands:
+  drop     Drop the database
+  create   Create the database
+  migrate  Migrate the database
+  reset    Reset (drop, create, migrate) the database
+  seed     Seed the database
+  help     Print this message or the help of the given subcommand(s)
```
Comment (BobrImperator) [inline README.md:194]: It'd be awesome to include those in the generated Readme if possible.

```suggestion
Commands:
  drop     Drop the database
  create   Create the database
  migrate  Migrate the database
  reset    Reset (drop, create, migrate) the database
  seed     Seed the database
  help     Print this message or the help of the given subcommand(s)
```

**Review by BobrImperator — COMMENTED** [review]
Ideas:
- `pace` without arguments should output `--help` by default
- It'd be awesome to include the commands and their explanation in generated project's README. Currently it only has a project's name.
- I think that the blueprint is missing a common place for starting other server dependencies. If I needed to create a background task, I'm forced to repeat the setup in both tests and the actual server. 


Questions:
- Would it be OK to add the `cli/` crate to `.gitignore`?
It causes quite a bit of bloat when searching up files.
Alternatively, we could probably communicate that this is something the users won't have to touch anymore and they're free to exclude it themselves.

**Comment by marcoow** [comment]
> It'd be awesome to include the commands and their explanation in generated project's README. Currently it only has a project's name.

Yes, that's to be done still

> I think that the blueprint is missing a common place for starting other server dependencies. If I needed to create a background task, I'm forced to repeat the setup in both tests and the actual server.

I think that's something for a later release. There's quite some questions coming with that, e.g. what do you use as a persistent queue for a background tasks implementation.

> Would it be OK to add the cli/ crate to .gitignore?

No since we want that to be checked in with the project. I understand you want to exclude it from your editor but I don't think adding it to `.gitignore` is the right way to achieve that.

File: README.md [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+```rs
+#[derive(Deserialize, Clone, Debug)]
+pub struct Config {
+    pub server: ServerConfig,
+    pub database: DatabaseConfig, // The database configuration only exists for projects that use a database
+    // add your config settings here…
+}
+```
+
+The values for the server and database configuration are read from the `APP_SERVER__IP`, `APP_SERVER__PORT`, and `APP_DATABASE__URL` environment variables. Any application-specific settings are first read from `app.toml` and then from an environment-specific file, e.g. `production.toml` so that environment-specific settings override those in `app.toml`. The main fils and folders in the crate are:
```
[the remark this answers is not in the document]
Comment (marcoow) [reply README.md:165]: @LukeMathWalker introduced that

File: README.md (hunk 2 above)
Comment (marcoow) [reply README.md:194]: The READMEs for the generated projects are to be done still

File: README.md (hunk 1 above)
Comment (marcoow) [reply README.md:198]: Yes, but I'd leave that for the READMEs of the generated projects.

File: README.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (LukeMathWalker) [reply README.md:165]: Standard mechanism to navigate to nested config parameters.

**Pull request merged**